Zend/zend_exceptions.c

```diff
@@ -772,21 +773,40 @@ void zend_register_default_exception(void) /* {{{ */
 		}
 	} ZEND_HASH_FOREACH_END();
 
-	INIT_CLASS_ENTRY(ce, "ErrorException", error_exception_functions);
-	error_exception_ce = zend_register_internal_class_ex(&ce, default_exception_ce);
-	error_exception_ce->create_object = zend_error_exception_new;
-	zend_declare_property_long(error_exception_ce, "severity", sizeof("severity")-1, E_ERROR, ZEND_ACC_PROTECTED);
+	INIT_CLASS_ENTRY(ce, "Error", NULL);
+	error_ce = zend_register_internal_class_ex(&ce, base_exception_ce);
+	error_ce->create_object = zend_default_exception_new;
+	zend_declare_property_long(error_ce, "severity", sizeof("severity")-1, E_ERROR, ZEND_ACC_PROTECTED);
 
-	INIT_CLASS_ENTRY(ce, "EngineException", NULL);
-	engine_exception_ce = zend_register_internal_class_ex(&ce, base_exception_ce);
+	/* A trick, to make visible private properties of BaseException */
+	ZEND_HASH_FOREACH_PTR(&error_ce->properties_info, prop) {
+		if (prop->flags & ZEND_ACC_SHADOW) {
+			if (prop->name->len == sizeof("\0BaseException\0string")-1) {
+				prop->flags &= ~ZEND_ACC_SHADOW;
+				prop->flags |= ZEND_ACC_PRIVATE;
+				prop->ce = error_ce;
+			} else if (prop->name->len == sizeof("\0BaseException\0trace")-1) {
+				prop->flags &= ~ZEND_ACC_SHADOW;
+				prop->flags |= ZEND_ACC_PRIVATE;
+				prop->ce = error_ce;
+			} else if (prop->name->len == sizeof("\0BaseException\0previous")-1) {
+				prop->flags &= ~ZEND_ACC_SHADOW;
+				prop->flags |= ZEND_ACC_PRIVATE;
+				prop->ce = error_ce;
+			}
+		}
+	} ZEND_HASH_FOREACH_END();
+
+	INIT_CLASS_ENTRY(ce, "EngineError", NULL);
+	engine_exception_ce = zend_register_internal_class_ex(&ce, error_ce);
 	engine_exception_ce->create_object = zend_default_exception_new;
 
-	INIT_CLASS_ENTRY(ce, "ParseException", NULL);
-	parse_exception_ce = zend_register_internal_class_ex(&ce, base_exception_ce);
+	INIT_CLASS_ENTRY(ce, "ParseError", NULL);
```

Comment:
Why name an exception with the "Error" wording instead of "Exception" ? It will be thrown as an exception. That may confuse people